src/GraphQL.Tests/Subscription/SubscriptionExecutionStrategyTests.cs

```diff
@@ -0,0 +1,678 @@
+#nullable enable
+#pragma warning disable CS1998 // Async method lacks 'await' operators and will run synchronously
+#pragma warning disable IDE0053 // Use expression body for lambda expressions
+
+using GraphQL.Execution;
+using GraphQL.MicrosoftDI;
+using GraphQL.Subscription;
+using GraphQL.Types;
+using Microsoft.Extensions.DependencyInjection;
+
+namespace GraphQL.Tests.Subscription;
+
+public class SubscriptionExecutionStrategyTests
+{
+    private SampleObservable<string> Source { get; } = new();
+    private SampleObserver? Observer { get; set; }
+    private IDisposable? Disposer { get; set; }
```

Comment:
```suggestion
    private IDisposable? Subscription { get; set; }
```

@Shane32 I'm going to finish review today. Two test files left + check `ObservableExtensions` one more time. You have done an excellent job 🥇 . I would like you to go on in next PRs with all that renamings stuff around subscriptions that we were talking about.

Reply:
I do plan to finish up the subscription stuff that is listed in #3009 - Remove SubscriptionExecutionResult, Add ScopedSubscriptionExecutionStrategy, and renaming subscription classes/interfaces/methods.

I am also working on reworking (improving) the server project, but I'm not sure how you will feel about my changes there - we need to discuss.

Reply:
`Subscription` has a naming conflict with some class or something.  I just renamed it to `SubscriptionObj`.

Reply:
ok